packages/calypso-products/src/is-difm-product.ts

```diff
@@ -0,0 +1,8 @@
+import { WPCOM_DIFM_LITE } from './constants';
+import { formatProduct } from './format-product';
+
+export function isDIFMProduct( product: any ): boolean {
```

Comment:
It's probably unlikely that `product` is random things; what about `product: { product_slug: string }`?

Reply:
Again I was assuming that we are uncertain whether the  cart item might have the property in snake case or camel case, which is why I though we used the formatProduct and which is why I did not strongly type the  parameter. Can we be sure the cart items  will be of a given case?

Reply:
Right now it does return the values in camel case,
![image](https://user-images.githubusercontent.com/3422709/132629659-ec51ea0a-9bad-40a8-8a6a-7cc314f2a709.png)

Maybe I can use some optional types like this ? WDYT 🤔 
`product: { product_slug?: string , productSlug?: string }`

Reply:
I changed the type to `product: { product_slug?: string , productSlug?: string }` just to be safe. I can iterate on this after the release today if you think its better to just remove the format product 👍 .

Reply:
That's fine, but I still think that it's best not to support `productSlug` because you're only using this function in one place and that place does not pass data in camelCase. When a consumer comes along that does need the camelCase version, they could expand the definition of this function, but until then using `formatProduct` is just premature optimization.